docetl/parsing_tools.py

```diff
@@ -0,0 +1,138 @@
+import os
+from typing import Optional, List
+from litellm import transcription
+
+
+def whisper_speech_to_text(filename: str) -> List[str]:
+    """
+    Transcribe speech from an audio file to text using Whisper model via litellm.
+    If the file is larger than 25 MB, it's split into 10-minute chunks with 30-second overlap.
+
+    Args:
+        filename (str): Path to the mp3 or mp4 file.
+
+    Returns:
+        List[str]: Transcribed text.
+    """
+    import os
+
+    file_size = os.path.getsize(filename)
+    if file_size > 25 * 1024 * 1024:  # 25 MB in bytes
+        from pydub import AudioSegment
+
+        audio = AudioSegment.from_file(filename)
+        chunk_length = 10 * 60 * 1000  # 10 minutes in milliseconds
+        overlap = 30 * 1000  # 30 seconds in milliseconds
+
+        chunks = []
+        for i in range(0, len(audio), chunk_length - overlap):
+            chunk = audio[i : i + chunk_length]
+            chunks.append(chunk)
+
+        transcriptions = []
+        for i, chunk in enumerate(chunks):
```

Comment:
some things that come to mind
- sequential (vs parallel) transcriptions of the chunk can be a problem depending on the number of chunks
- so, it seems like each chunk is written to disk and then immediately loaded back into memory. is it because `whisper-1` exclusively only deals with `mp3` audio file formats AND a chunk item (derived from an `AudioSegment`) can't be transcoded into `mp3` without doing a disk write?

Reply:
Good call. disk writing will slow it down considerably

Reply:
Good call. disk writing will slow it down considerably